server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -219,4 +248,114 @@ protected String contentType() {
         return CONTENT_TYPE;
     }
 
+    private static class TermVectorsValueFetcher implements ValueFetcher {
```

Comment:
Should this class be in a separate file?

Reply:
I renamed it SparseVectorValueFetcher since it's not a generic class. It's very specific to how the FeatureField works with a single position for each term so I think it's ok to keep it there.